frame/sudo/src/tests.rs

```diff
@@ -0,0 +1,67 @@
+use super::*;
+use mock::{
+	Sudo, Origin, Call, Test, PrivelegedCall, new_test_ext,
+}; 
+use frame_support::{assert_ok, assert_err};
+
+#[test]
+fn new_test_ext_and_sudo_get_key_works() {
+	// Test that the enivroment setup and pallets key retrieval works as expected
+	new_test_ext(1).execute_with(|| {
+		assert_eq!(Sudo::key(),  1u64);
+	});
+}
+
+#[test]
+fn sudo_basics() {
+	new_test_ext(1).execute_with(|| {
+		// A privelleged function should work when sudo is passed the root_key as origin
+		let call = Box::new(Call::Priveleged(PrivelegedCall::privileged_function()));
+		assert_ok!(Sudo::sudo(Origin::signed(1), call));
```

Comment:
Sudo will always return `Ok`, you need to check that your `privileged_function` succeeded.

As mentioned above, probably best to check via a state change, but you can also look at the event emitted to see the dispatch result.

Reply:
The test now use a logger which is called passed to relevant functions. The log is then checked to make sure the logger executed correctly.